### DISCUSSION TITLE
ThorVG and plain OpenGL

### DISCUSSION THREAD

**Discussion by GTSKA** [thorvg/thorvg#2299] [Q&A] [answered]
I'm trying to run ThorVG as simple as possible since EFL libraries are marked as malware by my firewall. However the most I get is a colored texture, not even a single form, trying to load svg also marks invalid argument errors. Are there examples without EFL libraries?

**Comment by hermet** [comment]
@GTSKA hello, probably, you can pick this patch and test? https://github.com/thorvg/thorvg/pull/2285 made by @onecoolx

**Reply by GTSKA** [reply]
seems pretty useful however i run Windows only, I'll give it a check anyway

**Comment by vtorri** [comment]
I have used the EFL with thorvg on Windows without problem.
thorvg provides a BGRA buffer (or similar colorspace). If you can display such buffer with your plain opengl application, there  will be no problem. You can use SDL in you application and fill an SDL_Texture with the buffer and it will work.
EFL are used only for examples, thorvg library does not depend on them

**Reply by GTSKA** [reply]
some examples are working now but specifically svg keeps telling that path is invalid

**Comment by hermet** [comment]
@GTSKA GLUT might be simpler than other toolkits. See this as well: https://github.com/mmaciola/thorvg-freeglut

**Reply by GTSKA** [reply]
got the same invalid argument error with that code, seems like thorvg can't load images on windows

**Reply by hermet** [reply]
@GTSKA If the resource is visible here (https://www.thorvg.org/viewer), then it should work properly. Please raise an issue here https://github.com/thorvg/thorvg/issues if you think it's a thorvg problem.

**Reply by GTSKA** [reply]
I'm using the same as the example you shared, it does load, also I've tried both relative and absolute path
Here is a zip with code and project files
[TestThorVG.zip](https://github.com/thorvg/thorvg/files/15392403/TestThorVG.zip)

**Reply by hermet** [reply]
@GTSKA hmm..  if it does load, which api comes up with the invalid argument?

**Reply by hermet** [reply]
Ok, examples might not working on windows now due to the toolkit portability issue. Instead you can refer src/tools/svg2png how to load a svg on windows. I assure it works.

**Reply by GTSKA** [reply]
Oh!! thanks a lot

**Reply by GTSKA** [reply]
tried that, still same result. Invalid Argument
No idea what else to do

**Comment by vtorri** [comment]
with EFL, it works on Windows...

**Reply by vtorri** [reply]
and the EFL are clearly not a malware

**Reply by GTSKA** [reply]
can't do much about company network policies, also I intend to run without EFL

**Comment by hermet** [comment, marked as answer]
Let's talk here: https://github.com/thorvg/thorvg/issues/2301
